Classes/common/CDTFetchChanges.m

```diff
@@ -79,12 +90,46 @@ - (void)main
                                                            options:&options
                                                             filter:nil
                                                             params:nil];
-        lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
-    } while (changes.count > 0);
+        if (!thereIsALimit) {
+            // Limit not set. Get as many results as we can.
+            lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
+
+            doLoop = (changes.count > 0);
+        } else if (mResultsLimit > 0) {
+            // There is a limit but we have not reached it yet.
+            lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
+
+            if (changes.count == 0) {
```

Comment:
That is true, I have:

```
lastSequence = [self notifyChanges:changes startingSequence:lastSequence];
```

In 2 branches but not in the [third one](https://github.com/cloudant/CDTDatastore/pull/177/files#diff-bac15dfde454859cd8e3a8dc04fcf29fR127) where I only update `moreComing`.

About:

```
doLoop = (changes.count > 0);
```

And

```
if (changes.count == 0) {
  doLoop = NO;
}
```

The `if` right above also has an [else](https://github.com/cloudant/CDTDatastore/pull/177/files#diff-bac15dfde454859cd8e3a8dc04fcf29fR109) where I prepare the `.limit` for the next loop.